crates/conjure_rules_proc_macro/src/lib.rs

```diff
@@ -0,0 +1,65 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, Ident, ItemFn};
+
+#[doc(hidden)]
+#[proc_macro_attribute]
+/// This procedural macro registers a decorated function with `conjure_rules`' global registry.
+/// It may be used in any downstream crate. For more information on linker magic, see the [`linkme`](https://docs.rs/linkme/latest/linkme/) crate.
+///
```

Comment:
(Pedantic) we should treat this crate as internal to `conjure_rules`, so no-one else should use / see this but us

Reply:
The macro itself will be externally visible if `conjure_rules` is imported, no?

Reply:
through `conjure_rules` only :)

Reply:
unless the user explicitly imports `conjure_rules_proc_macro`, which we should tell them not to do in the description for conjure_rules_proc-macro.

Reply:
btw, you can directly document the re-export in `conjure_rules`, you dont need to put the docs here.